cli/command/stack/deploy.go

```diff
@@ -19,20 +25,28 @@ func newDeployCommand(dockerCli command.Cli, common *commonOptions) *cobra.Comma
 		Args:    cli.ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
 			opts.Namespace = args[0]
+			if err := validateStackName(opts.Namespace); err != nil {
+				return err
+			}
+
 			switch {
-			case common == nil: // Top level deploy commad
-				return swarm.RunDeploy(dockerCli, opts)
-			case common.orchestrator.HasAll():
-				return errUnsupportedAllOrchestrator
-			case common.orchestrator.HasKubernetes():
-				kli, err := kubernetes.WrapCli(dockerCli, kubernetes.NewOptions(cmd.Flags(), common.orchestrator))
-				if err != nil {
-					return err
-				}
-				return kubernetes.RunDeploy(kli, opts)
-			default:
-				return swarm.RunDeploy(dockerCli, opts)
+			case opts.Bundlefile == "" && len(opts.Composefiles) == 0:
+				return errors.Errorf("Please specify either a bundle file (with --bundle-file) or a Compose file (with --compose-file).")
+			case opts.Bundlefile != "" && len(opts.Composefiles) != 0:
+				return errors.Errorf("You cannot specify both a bundle file and a Compose file.")
+			case opts.Bundlefile != "":
+				return swarm.DeployBundle(context.Background(), dockerCli, opts)
```

Comment:
we should produce an error if the current orchestrator is kubernetes (as we don't support bundle files on kube, correct?)